director/projects/api/views/snapshots.py

```diff
@@ -0,0 +1,258 @@
+import os
+
+from django.conf import settings
+from django.db import IntegrityError
+from django.http import HttpResponse, FileResponse
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import (
+    negotiation,
+    mixins,
+    permissions,
+    status,
+    viewsets,
+)
+from rest_framework.decorators import action
+from rest_framework.exceptions import PermissionDenied, NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from general.api.exceptions import ConflictError
+from lib.converter_facade import (
+    ConverterFacade,
+    ConverterIo,
+    ConverterIoType,
+    ConverterContext,
+)
+from lib.conversion_types import ConversionFormatId
+from lib.data_cleaning import logged_in_or_none
+from projects.models import Snapshot
+from projects.project_models import ProjectEvent, ProjectEventType, ProjectEventLevel
+from projects.snapshots import ProjectSnapshotter, SnapshotInProgressError
+from projects.views.mixins import ProjectPermissionsMixin, ProjectPermissionType
+from projects.api.serializers import SnapshotSerializer, SnapshotCreateRequestSerializer
+from projects.source_operations import snapshot_path
+
+
+class SnapshotsViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+    ProjectPermissionsMixin,
+):
+
+    # Settings
+
+    serializer_class = SnapshotSerializer
+    lookup_field = "number"
+
+    def get_permissions(self):
+        """
+        Get the list of permissions that the current action requires.
+
+        - `list` and `retrieve`: auth not required but access denied for private projects
+        - `create`: auth required
+        """
+        if self.action == "create":
```

Comment:
It feels like this method is trying to be overly clever. 

Just do something like `return [permissions.IsAuthenticated] is self.action == "create" else []`